src/gpgpu-sim/gpu-cache.cc

```diff
@@ -279,20 +279,27 @@ enum cache_request_status tag_array::probe(new_addr_type addr, unsigned &idx,
       }
     }
     if (!line->is_reserved_line()) {
-      all_reserved = false;
-      if (line->is_invalid_line()) {
-        invalid_line = index;
-      } else {
-        // valid line : keep track of most appropriate replacement candidate
-        if (m_config.m_replacement_policy == LRU) {
-          if (line->get_last_access_time() < valid_timestamp) {
-            valid_timestamp = line->get_last_access_time();
-            valid_line = index;
-          }
-        } else if (m_config.m_replacement_policy == FIFO) {
-          if (line->get_alloc_time() < valid_timestamp) {
-            valid_timestamp = line->get_alloc_time();
-            valid_line = index;
+      if (!line->is_modified_line() ||
+          100 * m_dirty / (m_config.m_nset * m_config.m_assoc) >=
```

Comment:
rewrite this to be  ( m_dirty * 100)

Reply:
sorry or rewrite it as  (m_dirty / (m_config.m_nset * m_config.m_assoc)) * 100, this is better to undersantd